### PR TITLE
Add non-standard data type serialization mechanism

### PR DESCRIPTION
**Pull request by yuslepukhin**
**Description**
This PR enables us to serialize datatypes that at are not ONNX datatypes.

**Motivation and Context**

AutoML will introduce a few new datatypes such as Table that are eventually will become candidates to ONNX standardization. However, the standardization process sometimes takes too long and we want to be able to test models that have ONNX-like automl operators that are not yet standard and operate on our custom datatypes and have a proof of concept.

This change enables us to introduce new datatypes into a our local copy of the protobuf and save serialized models under the disguise of Opaque datatypes so we can testrun those models and compare expected outputs with the actual output.s

This PR still makes use of ONNX interned strings such as opaqu(domain_string, name_string) but it intercepts string to TypeProto and vice versa mapping requests so we put our custom datatype proto and our MLDataType where appropriate.

### CODE REVIEW

File: onnxruntime/core/protobuf/onnx.in.proto [hunk 1]
```diff
@@ -535,6 +554,8 @@ message TypeProto {
 
     SparseTensor sparse_tensor_type = 8;
 
+    TimePoint    timepoint_type = 100;
```
Comment (snnn) [inline onnxruntime/core/protobuf/onnx.in.proto]: This struct is for describing the type(input/output type), not for serialization. TensorProto is the corresponding one for serialization. As you can see, Tensor and TensorProto are different.

Comment (snnn) [reply onnxruntime/core/protobuf/onnx.in.proto]: What I can see, TimePoint is more like the TensorProto, not the Tensor.

Comment (yuslepukhin) [reply onnxruntime/core/protobuf/onnx.in.proto]: Yes, it is for input/output

---
In reply to: [315352438](https://github.com/microsoft/onnxruntime/pull/1649#discussion_r315352438) [](ancestors = 315352438)

Comment (yuslepukhin) [reply onnxruntime/core/protobuf/onnx.in.proto]: Why can I not re-use the same definition say for tests input or expected output?

---
In reply to: [315352704](https://github.com/microsoft/onnxruntime/pull/1649#discussion_r315352704) [](ancestors = 315352704)

Comment (snnn) [reply onnxruntime/core/protobuf/onnx.in.proto]: The TypeProto  is for describing the type, not the value.  

BTW, is TimePoint  is an opaque type or not?

Comment (yuslepukhin) [reply onnxruntime/core/protobuf/onnx.in.proto]: Okay, thanks for clarification. Yes, currently I am masqerading under Opaque type but in the future such types would become on their own.

---
In reply to: [315357164](https://github.com/microsoft/onnxruntime/pull/1649#discussion_r315357164) [](ancestors = 315357164)

**Comment by yuslepukhin** [comment]
I will need to rework this.

**Pull request closed without merging**